internal/app/tfsec/aws_no_ecr_repository_scan_test.go

```diff
@@ -0,0 +1,57 @@
+package tfsec
+
+import (
+	"testing"
+
+	"github.com/liamg/tfsec/internal/app/tfsec/scanner"
+
+	"github.com/liamg/tfsec/internal/app/tfsec/checks"
+)
+
+func Test_AWSEcrImageScanNotEnabled(t *testing.T) {
+	var tests = []struct {
+		name                  string
+		source                string
+		mustIncludeResultCode scanner.RuleID
+		mustExcludeResultCode scanner.RuleID
+	}{
+		{
+			name:                  "check KMS Key with auto-rotation not set",
+			source:                `resource "aws_ecr_repository" "foo" {}`,
+			mustIncludeResultCode: checks.AWSEcrImageScanNotEnabled,
+		},
+		{
+			name: "check ECR Image Scan disabled",
+			source: `
+resource "aws_ecr_repository" "foo" {
+  name                 = "bar"
+  image_tag_mutability = "MUTABLE"
+
+  image_scanning_configuration {
+    scan_on_push = false
+  }
+}`,
+			mustIncludeResultCode: checks.AWSEcrImageScanNotEnabled,
+		},
+		{
+			name: "check ECR Image Scan disabled",
+			source: `
+resource "aws_ecr_repository" "foo" {
+  name                 = "bar"
+  image_tag_mutability = "MUTABLE"
+
+  image_scanning_configuration {
+    scan_on_push = false
```

Comment:
This test should include `scan_on_push = true` as it ensures there are no false positives.
```suggestion
    scan_on_push = true
```